packages/ref-core/src/cmip_ref_core/providers.py

```diff
@@ -86,3 +99,176 @@ def get(self, slug: str) -> Metric:
             The requested metric.
         """
         return self._metrics[slug.lower()]
+
+
+def import_provider(fqn: str) -> MetricsProvider:
+    """
+    Import a provider by name
+
+    Parameters
+    ----------
+    fqn
+        Full package and attribute name of the provider to import
+
+        For example: `cmip_ref_metrics_example.provider` will use the `provider` attribute from the
+        `cmip_ref_metrics_example` package.
+
+        If only a package name is provided, the default attribute name is `provider`.
+
+    Raises
+    ------
+    InvalidProviderException
+        If the provider cannot be imported
+
+        If the provider isn't a valid `MetricsProvider`.
+
+    Returns
+    -------
+    :
+        MetricsProvider instance
+    """
+    if "." in fqn:
+        module, name = fqn.rsplit(".", 1)
+    else:
+        module = fqn
+        name = "provider"
+
+    try:
+        imp = importlib.import_module(module)
+        provider = getattr(imp, name)
+        provider.module = imp  # TODO: fix this hack
+        if not isinstance(provider, MetricsProvider):
+            raise InvalidProviderException(fqn, f"Expected MetricsProvider, got {type(provider)}")
+        return provider
+    except ModuleNotFoundError:
+        logger.error(f"Module '{fqn}' not found")
+        raise InvalidProviderException(fqn, f"Module '{module}' not found")
+    except AttributeError:
+        logger.error(f"Provider '{fqn}' not found")
+        raise InvalidProviderException(fqn, f"Provider '{name}' not found in {module}")
+
+
+class CommandLineMetricsProvider(MetricsProvider):
+    """
+    A metrics provider for metrics that can be run from the command line.
+    """
+
+    @abstractmethod
+    def run(self, cmd: Iterable[str]) -> None:
+        """
+        Return the command to run.
+        """
+
+
+MICROMAMBA_EXE = (
+    "https://github.com/mamba-org/micromamba-releases/releases/latest/download/micromamba-linux-64"
+)
+
+
+class CondaMetricsProvider(CommandLineMetricsProvider):
+    """
+    A provider for metrics that can be run from the command line in a conda environment.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        version: str,
+        slug: str | None = None,
+    ) -> None:
+        super().__init__(name, version, slug)
+        self._conda_exe: Path | None = None
```

Comment:
Should we check an environment variable if a particular system conda should be used and fallback to installing micromamba if needed?

It looks like this always installs micromamba. Is it a pretty safe assumption that micromamba will work in an HPC setting? I haven't used it before

An inspection of my env variables suggests that `CONDA_EXE` if conda is installed even if no base environment is activated. I do have to admit I don't use conda too much so this is somewhat outside my wheelhouse.

```suggestion
        self._conda_exe: Path | None = env.path("CONDA_EXE", default=None)
```

with an `from cmip_ref_core.env import env` above

Reply:
I think it is more reliable to install our own. Outdated conda versions are a source of installation issues and the file is pretty small compared to the size of the conda environments.